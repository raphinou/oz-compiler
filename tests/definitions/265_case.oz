% cases with captures and guards
local
   R=lab(a b c)
   R2=rec(inner:R b c)
in
   case R
   of lab(a B ...) andthen B==b then
      {Show openrec}
      {Show B}
   [] lab(a b c) then
      {Show completerec}
   else
      {Show elseBranch}
   end

   case R
   of lab(a B ...) andthen B==c then
      {Show openrec}
      {Show B}
   [] lab(a C B) then % intentionally switched variable names
      {Show completerec}
      {Show C}
      {Show B}
   else
      {Show elseBranch}
   end

   case false
   of lab(a B ...) andthen B==c then
      {Show openrec}
      {Show B}
   [] lab(a C B) then
      {Show completerec}
      {Show C}
      {Show B}
   else
      {Show elseBranch}
   end

   case R2
   of false then
      {Show false}
   [] rec(inner:lab(Capture ...) ...) then
      {Show nested}
      {Show Capture}
   end
end
