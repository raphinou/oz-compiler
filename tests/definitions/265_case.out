openrec
b
completerec
b
c
elseBranch
nested
a
